govuk_chat_evaluation/rag_answers/evaluate.py

```diff
@@ -1,31 +1,123 @@
-def run_deepeval_evaluation(models: list[EvaluationTestCase]):
-    llm_test_cases = [model.to_llm_test_case() for model in models]
+    models = jsonl_to_models(evaluation_data_path, EvaluationTestCase)
 
-    # contrived example metric
-    correctness_metric = GEval(
-        name="Correctness",
-        criteria="Determine if the 'actual output' is correct based on the 'expected output'.",
-        evaluation_params=[
-            LLMTestCaseParams.ACTUAL_OUTPUT,
-            LLMTestCaseParams.EXPECTED_OUTPUT,
-        ],
-        threshold=0.5,
+    evaluation_outputs = run_deepeval_evaluation(
+        cases=[model.to_llm_test_case() for model in models],
+        metrics=cast(list[BaseMetric], evaluation_config.get_metric_instances()),
+        n_runs=evaluation_config.n_runs,
+        **DEEPEVAL_EVAL_PARAMETERS
     )
 
-    deepeval_evaluate(llm_test_cases, [correctness_metric], print_results=False)  # type: ignore
+    evaluation_results = convert_deepeval_output_to_evaluation_results(evaluation_outputs)
+    
+    aggregation = AggregatedResults(evaluation_results)  # type: ignore
+
+    # calculate aggregated results and exports results to CSV files
+    aggregation.export_to_csvs(output_dir, prefix="alessia_test")
+
+    print("Evaluation Results:")
+    print(aggregation.summary)
+
+
+
+class AggregatedResults:
+    def __init__(
+        self,
+        evaluation_results: list[EvaluationResult]
+    ):
+        """
+        Args:
+            evaluation_results: List of TestResult objects.
+        """
+        self.evaluation_results = evaluation_results
+
+    @cached_property
+    def per_input_metric_averages(self) -> pd.DataFrame:
+        """
+        Computes average metric scores per test input.
+
+        Returns:
+            DataFrame with rows as test names and columns as metrics.
+        """
+        # flatten the evaluation results
+        data = []
+
+        for eval_result in self.evaluation_results:
+            for evaluation_output in eval_result.evaluation_results:
+                data.append({
+                    "name": eval_result.name,
+                    "input": eval_result.input,
+                    "metric": evaluation_output.metric,
+                    "score": evaluation_output.score,
+                })
+
+        df = pd.DataFrame(data)
+
+        return df.groupby(["name", "input", "metric"])["score"].agg(['mean', 'std']).unstack().reset_index()
+
+    @cached_property
+    def summary(self) -> pd.DataFrame:
+        """
+        Summary statistics across all inputs: median, mean, std per metric.
+
+        Returns:
+            DataFrame with metric as index and stats as columns.
+        """
+
+        mean_df = self.per_input_metric_averages["mean"]
+
+        return pd.DataFrame({
+            "median": mean_df.median(),
+            "mean": mean_df.mean(),
+            "std": mean_df.std(),
+        })
+
+    def export_to_csvs(self, _output_dir: Path, prefix: str = "metrics") -> None:
```

Comment:
we should remove the underscore prefix from output_dir here as it's a used argument.

I don't think we really need the prefix argument. I see you've used it in your example one but in normal circumstance an output dir is created per run so it seems very unlikely we'd need to prefix it.

Reply:
Good point. I also did not realise I had left an hardcoded prefix in the `evaluate_and_output_results` function... all fixed.